src/main/cpp/src/get_json_object.cu

```diff
@@ -38,40 +38,20 @@
 #include <rmm/device_uvector.hpp>
 #include <rmm/exec_policy.hpp>
 
-#include <thrust/optional.h>
-#include <thrust/pair.h>
-#include <thrust/scan.h>
-#include <thrust/tuple.h>
-
 namespace spark_rapids_jni {
 
 namespace detail {
-// namespace {
-
-/**
- * write JSON style
- */
-enum class write_style { raw_style, quoted_style, flatten_style };
-
-/**
- * path instruction type
- */
-enum class path_instruction_type { subscript, wildcard, key, index, named };
 
 /**
- * path instruction
+ * @brief Result of calling a parse function.
+ *
+ * The primary use of this is to distinguish between "success" and
+ * "success but no data" return cases.  For example, if you are reading the
+ * values of an array you might call a parse function in a while loop. You
+ * would want to continue doing this until you either encounter an error
+ * (parse_result::ERROR) or you get nothing back (parse_result::EMPTY)
```

Comment:
But there is no parse_result::EMPTY in the enum?

Reply:
Same question.

Reply:
Removed `parse_result`
We do not need `parse_result::EMPTY`.
If `evaluate_path` success and result is empty, sets `output_size` as 0 and `evaluate_path` returns true.